hazelcast/src/main/java/com/hazelcast/internal/serialization/SerializationServiceBuilder.java

```diff
@@ -63,5 +64,7 @@
 
     SerializationServiceBuilder setInitialOutputBufferSize(int initialOutputBufferSize);
 
+    SerializationServiceBuilder setClusterService(ClusterService clusterService);
```

Comment:
What about registering the `SerializationService` as a cluster version listener? (not part of this PR); it would require some listener support code on the `ClusterStateManager` side, but should reduce coupling.